osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/virtualizationconnector/VCDeleteMetaTask.java

```diff
@@ -57,20 +60,27 @@ public void executeTransaction(EntityManager em) throws Exception {
             OSCEntityManager<VirtualizationConnector> vcEntityMgr = new OSCEntityManager<>(VirtualizationConnector.class, em, this.txBroadcastUtil);
             VirtualizationConnector connector = vcEntityMgr.findByPrimaryKey(this.vc.getId());
 
+            cleanKeystoneProviderConnection(connector);
+
             SslCertificateAttrEntityMgr sslCertificateAttrEntityMgr = new SslCertificateAttrEntityMgr(em, this.txBroadcastUtil);
             sslCertificateAttrEntityMgr.removeCertificateList(connector.getSslCertificateAttrSet());
             vcEntityMgr.delete(this.vc.getId());
         } catch (Exception ex) {
             throw ex;
-        } finally{
-        	 // Unlock VC.
+        } finally {
+            // Unlock VC.
             if (vcUnlockTask != null) {
                 log.info("Releasing lock for VC '" + this.vc.getName() + "'");
                 LockManager.getLockManager().releaseLock(new LockRequest(vcUnlockTask));
             }
         }
     }
 
+    private void cleanKeystoneProviderConnection(VirtualizationConnector vc) throws EncryptionException {
```

Comment:
It's solution for manipulating map as there is no overall way for clearing the map (only when application is turned down) so it basically remove used existing endpoint if it's removed from OSC or updated. In rest case i don't see place where we could perform cleaning whole map as always some jobs could use connection.